Azure/ARO-RP
# New cmd to populate PlatformWorkloadIdentityRoleSets in dev and prod

### Which issue this PR addresses:


Fixes https://issues.redhat.com/browse/ARO-6449

### What this PR does / why we need it:


Adds a new cmd for populating our new proxy resource (PlatformWorkloadIdentityRoleSetDocument) with role sets in prod, and adds the hack code to populate it in dev too. This resource is used to determine identity requirements for supported OCP versions.

### Test plan for issue:

- [x] Run a dev RP ARO install using these changes, confirm that the role set is populated in cosmosDB
- [x] Confirm prod script works using local environment variables set
  - Steps to test:
  1) `export PLATFORM_WORKLOAD_IDENTITY_ROLE_SETS='[{"openShiftVersion":"4.14","platformWorkloadIdentityRoles":[{"operatorName":"CloudControllerManager","roleDefinitionName":"Azure RedHat OpenShift Cloud Controller Manager Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/a1f96423-95ce-4224-ab27-4e3dc72facd4","serviceAccounts":["openshift-cloud-controller-manager:cloud-controller-manager"]},{"operatorName":"ClusterIngressOperator","roleDefinitionName":"Azure RedHat OpenShift Cluster Ingress Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/0336e1d3-7a87-462b-b6db-342b63f7802c","serviceAccounts":["openshift-ingress-operator:ingress-operator"]},{"operatorName":"MachineApiOperator","roleDefinitionName":"Azure RedHat OpenShift Machine API Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/0358943c-7e01-48ba-8889-02cc51d78637","serviceAccounts":["openshift-machine-api:machine-api-operator"]},{"operatorName":"StorageOperator","roleDefinitionName":"Azure RedHat OpenShift Storage Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/5b7237c5-45e1-49d6-bc18-a1f62f400748","serviceAccounts":["openshift-cluster-csi-drivers:azure-disk-csi-driver-operator","openshift-cluster-csi-drivers:azure-disk-csi-driver-controller-sa"]},{"operatorName":"NetworkOperator","roleDefinitionName":"Azure RedHat OpenShift Network Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/be7a6435-15ae-4171-8f30-4a343eff9e8f","serviceAccounts":["openshift-cloud-network-config-controller:cloud-network-config-controller"]},{"operatorName":"ImageRegistryOperator","roleDefinitionName":"Azure RedHat OpenShift Image Registry Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/8b32b316-c2f5-4ddf-b05b-83dacd2d08b5","serviceAccounts":["openshift-image-registry:cluster-image-registry-operator","openshift-image-registry:registry"]},{"operatorName":"AzureFilesStorageOperator","roleDefinitionName":"Azure RedHat OpenShift Azure Files Storage Operator Role","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/0d7aedc0-15fd-4a67-a412-efad370c947e","serviceAccounts":["openshift-cluster-csi-drivers:azure-file-csi-driver-operator","openshift-cluster-csi-drivers:azure-file-csi-driver-controller-sa","openshift-cluster-csi-drivers:azure-file-csi-driver-node-sa"]},{"operatorName":"ServiceOperator","roleDefinitionName":"Azure RedHat OpenShift Service Operator","roleDefinitionId":"/providers/Microsoft.Authorization/roleDefinitions/4436bae4-7702-4c84-919b-c4069ff25ee2","serviceAccounts":["openshift-azure-operator:aro-operator-master"]}]}]'`
  2) Source regular env vars, and then run `go run ./cmd/aro update-role-sets` and observe that the role set above is populated.


### Is there any documentation that needs to be updated for this PR?



### How do you know this will function as expected in production?
